File: quicksort/quick_sort.py
#!/usr/bin/python

array = [2,2,1,3,4,5,6,300,24,7,11]

def quickSort(list):
    if len(list) > 1:
        llist = []
        rlist = []
        key = [list[0]]
        for i in range(1, len(list)):
            if list[i] > key[0]:
	        rlist.append(list[i])
	    elif list[i] < key[0]:
                llist.append(list[i])
	    else:
	        key.append(list[i])
        return quickSort(llist) + key + quickSort(rlist)
    else:
        return list

if __name__ == "__main__":
    print "before quick sort:", array
    print "after quick sort:", quickSort(array)   

File: quicksort/quicksort.go
package main

import "fmt"

func main() {
	list := []int{1, 3, 4, 5, 6, 100, 43, 3}
	fmt.Println("before quick sort:", list)
	fmt.Println("after quick sort:", QuickSort(list))
}

func QuickSort(list []int) []int {
	if len(list) > 1 {
		rlist := make([]int, 0)
		llist := make([]int, 0)
		key := make([]int, 0)
		key = append(key, list[0])
		for i := 1; i < len(list); i++ {
			if list[i] < key[0] {
				llist = append(llist, list[i])
			} else if list[i] > key[0] {
				rlist = append(rlist, list[i])
			} else {
				key = append(key, list[i])
			}
		}
		return append(QuickSort(llist), append(key, QuickSort(rlist)...)...)
	} else {
		return list
	}
}
